src/layer/tile/GridLayer.js

```diff
@@ -356,13 +361,17 @@ L.GridLayer = L.Layer.extend({
 			this._updateLevels();
 			this._resetGrid();
 
-			if (!L.Browser.mobileWebkit) {
+			if (tileZoom !== undefined) {
 				this._update(center, tileZoom);
 			}
 
 			if (!noPrune) {
 				this._pruneTiles();
 			}
+
+			// Flag to prevent _updateOpacity from pruning tiles during
+			// a zoom anim or a pinch gesture
+			this._noPrune = !!noPrune;
```

Comment:
During a pinch gesture, if the user zooms in, the tile the event started on might be destroyed because `_onTileLoad` will call `_updateOpacity` which will call `_pruneTiles`, which checks parent/child tiles and prunes them if needed.